pipelines.go

```diff
@@ -24,15 +31,39 @@ func applyPipeline(c *cli.Context) error {
 		GetColoredText(filePath, color.FgCyan))
 	var content, _ = ReadFromFile(c.String("file"))
 
+	if yamlHasDockerUsername(content) {
+		if dockerUsername == "" || dockerUsername == defaults.DOCKER_USERNAME_PLACEHOLDER {
+			dockerUsername = TextInput("Enter a valid docker username:")
+		}
+		content = ReplacePlaceholderValues(content, defaults.DOCKER_USERNAME_PLACEHOLDER, dockerUsername)
+	}
+	if yamlHasGithubUsername(content) {
+		if githubUsername == "" || githubUsername == defaults.GITHUB_USERNAME_PLACEHOLDER {
+			githubUsername = TextInput("Enter valid github username:")
+		}
+		content = ReplacePlaceholderValues(content, defaults.GITHUB_USERNAME_PLACEHOLDER, githubUsername)
+	}
 	requestBody := GetJsonFromYaml(content)
 
 	if requestBody == nil {
 		println(GetColoredText("Please enter valid pipeline yaml file", color.FgRed))
 	}
 
 	identifier := ValueToString(GetNestedValue(requestBody, "pipeline", "identifier").(string))
-	projectIdentifier := ValueToString(GetNestedValue(requestBody, "pipeline", "projectIdentifier").(string))
-	orgIdentifier := ValueToString(GetNestedValue(requestBody, "pipeline", "orgIdentifier").(string))
+	if orgIdentifier != "" {
+		content = ReplacePlaceholderValues(content, defaults.DEFAULT_ORG, orgIdentifier)
+	} else {
+		orgIdentifier = ValueToString(GetNestedValue(requestBody, "pipeline", "orgIdentifier").(string))
+	}
+	if projectIdentifier == "" {
+		projectIdentifier = ValueToString(GetNestedValue(requestBody, "connector", "projectIdentifier").(string))
```

Comment:
```suggestion
		projectIdentifier = ValueToString(GetNestedValue(requestBody, "pipeline", "projectIdentifier").(string))
```
Should be pipeline.projectIdentifier, not connector.

Reply:
Current behavior:

```
Trying to create or update pipeline using the yaml= prpipeline.yaml
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
main.applyPipeline(0x400000c7f8?)
	/home/ubuntu/harness-cli/pipelines.go:59 +0x1218
main.cliWrapper(0x40000c2a01?, 0x400009ad00?)
	/home/ubuntu/harness-cli/main.go:635 +0x94
main.main.func26(0x4000130160?)
	/home/ubuntu/harness-cli/main.go:580 +0x28
github.com/urfave/cli/v2.(*Command).Run(0x4000130160, 0x400009ad00, {0x40001400b0, 0x1, 0x1})
	/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.25.5/command.go:274 +0x730
github.com/urfave/cli/v2.(*Command).Run(0x4000130000, 0x400009ab80, {0x400009abc0, 0x4, 0x4})
	/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.25.5/command.go:267 +0x940
github.com/urfave/cli/v2.(*Command).Run(0x4000130840, 0x400009a880, {0x400001e0a0, 0x5, 0x5})
	/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.25.5/command.go:267 +0x940
github.com/urfave/cli/v2.(*App).RunContext(0x400012c000, {0x4b0fb0?, 0x76c420}, {0x400001e0a0, 0x5, 0x5})
	/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.25.5/app.go:332 +0x518
github.com/urfave/cli/v2.(*App).Run(...)
	/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.25.5/app.go:309
main.main()
	/home/ubuntu/harness-cli/main.go:622 +0x3ca8
```